packages/fhir-team-management/src/components/TeamsAddEdit/tests/Form.test.tsx

```diff
@@ -0,0 +1,306 @@
+          <Form
+            fhirbaseURL={fhirBaseURL}
+            allPractitioner={practitioner.entry.map((e) => e.resource)}
+            allPractitionerRole={practitionerrole.entry.map((e) => e.resource)}
+          />
+        </QueryClientProvider>
+      </Router>
+    );
+
+    expect(wrapper.find('form')).toHaveLength(1);
+  });
+
+  it('renders without crashing with id', () => {
+    const queryClient = new QueryClient();
+    const wrapper = mount(
+      <Router history={history}>
+        <QueryClientProvider client={queryClient}>
+          <Form
+            fhirbaseURL={fhirBaseURL}
+            allPractitioner={practitioner.entry.map((e) => e.resource)}
+            allPractitionerRole={practitionerrole.entry.map((e) => e.resource)}
+            initialValue={TeamValue}
+          />
+        </QueryClientProvider>
+      </Router>
+    );
+
+    expect(wrapper.find('Form').prop('initialValue')).toMatchObject(TeamValue);
+    expect(wrapper.find('form')).toHaveLength(1);
+  });
+
+  it('Cancel button', () => {
+    const historyback = jest.spyOn(history, 'goBack');
+    const queryClient = new QueryClient();
+    const wrapper = mount(
+      <Router history={history}>
+        <QueryClientProvider client={queryClient}>
+          <Form
+            fhirbaseURL={fhirBaseURL}
+            allPractitioner={practitioner.entry.map((e) => e.resource)}
+            allPractitionerRole={practitionerrole.entry.map((e) => e.resource)}
+            initialValue={TeamValue}
+          />
+        </QueryClientProvider>
+      </Router>
+    );
+
+    expect(wrapper.find('form')).toHaveLength(1);
+    wrapper.find('button#cancel').simulate('click');
+    expect(historyback).toBeCalled();
+  });
+
+  it('Create Team', async () => {
+    const mockSuccessNotification = jest.spyOn(notifications, 'sendSuccessNotification');
+    const thenfn = jest.fn();
+    const catchfn = jest.fn();
+
+    onSubmit(
+      fhirBaseURL,
+      FormintialValue,
+      TeamValue,
+      practitioner.entry.map((e) => e.resource),
+      practitionerrole.entry.map((e) => e.resource)
+    )
+      .then(thenfn)
+      .catch(catchfn);
+
+    await act(async () => {
+      await flushPromises();
+    });
+
+    expect(thenfn).toBeCalled();
+    expect(catchfn).not.toBeCalled();
+    expect(mockSuccessNotification).toHaveBeenNthCalledWith(1, 'Successfully Added Teams');
+    expect(mockSuccessNotification).toHaveBeenNthCalledWith(
+      2,
+      'Successfully Assigned Practitioners'
+    );
+  });
+
+  it('Edit Team', async () => {
+    const mockSuccessNotification = jest.spyOn(notifications, 'sendSuccessNotification');
+    const thenfn = jest.fn();
+    const catchfn = jest.fn();
+    onSubmit(
+      fhirBaseURL,
+      TeamValue,
+      { ...TeamValue, name: 'new name', practitioners: ['116', '104'] },
+      practitioner.entry.map((e) => e.resource),
+      practitionerrole.entry.map((e) => e.resource)
+    )
+      .then(thenfn)
+      .catch(catchfn);
```

Comment:
would love to see an integration-ish test where we simulate a user adding values to the form and clicking submit instead of just calling the onSubmit function manually,

Reply:
alternatively i updated another test which simulates user updating fields https://github.com/opensrp/web/blob/0b23abbb3f27fe71716909320a455ae85dc02314/packages/fhir-team-management/src/components/TeamsAddEdit/tests/Form.test.tsx#L222-L231